data-products/src/main/scala/org/sunbird/obsrv/dataproducts/job/MasterDataProcessorIndexer.scala

```diff
@@ -0,0 +1,211 @@
+      case "local" => "file"
+      case "aws" => "s3"
+      case "azure" => "azure"
+      case "gcloud" => "gs"
+      case "cephs3" => "s3" // TODO: Have to check Druid compatibility
+      case "oci" => "s3" // TODO: Have to check Druid compatibility
+      case _ => throw new Exception("Unsupported provider")
+    }
+  }
+
+  @throws[Exception]
+  private def getProvider(): String = {
+    config.getString("cloudStorage.provider") match {
+      case "local" => "file"
+      case "aws" => "s3"
+      case "azure" => "azure"
+      case "gcloud" => "google"
+      case "cephs3" => "s3" // TODO: Have to check Druid compatibility
+      case "oci" => "s3" // TODO: Have to check Druid compatibility
+      case _ => throw new Exception("Unsupported provider")
+    }
+  }
+
+  private def upload(cloudProvider: String) = {
+    cloudProvider match {
+      case "aws" =>
+        (bucketName: String, key: String, payload: File) => {
+
+          val s3Client = S3Client.builder()
+            .region(Region.US_EAST_2)
+            .credentialsProvider(DefaultCredentialsProvider.create())
+            .build()
+
+          val request = PutObjectRequest.builder()
+            .bucket(bucketName)
+            .key(key)
+            .contentType("application/json")
+            .build()
+
+          val jsonPayload = RequestBody.fromFile(payload)
+          s3Client.putObject(request, jsonPayload)
+        }
+      case _ => throw new Exception("Unsupported provider")
+    }
+  }
+
+  private def submitIngestionTask(ingestionSpec: String) = {
+    // TODO: Handle success and failure responses properly
+    val response = Unirest.post(config.getString("druid.indexer.url"))
+      .header("Content-Type", "application/json")
+      .body(ingestionSpec).asJson()
+    response.ifFailure(response => throw new Exception("Exception while submitting ingestion task"))
+  }
+
+  private def updateDataSourceRef(datasource: DataSource, datasourceRef: String): Unit = {
+    DatasetRegistry.updateDatasourceRef(datasource, datasourceRef)
+  }
+
+  private def deleteDataSource(datasourceRef: String): Unit = {
+    // TODO: Handle success and failure responses properly
+    val response = Unirest.delete(config.getString("druid.datasource.delete.url") + datasourceRef)
+      .header("Content-Type", "application/json")
+      .asJson()
+    response.ifFailure(response => throw new Exception("Exception while deleting datasource" + datasourceRef))
+  }
+
+  def createDataFile(dataset: Dataset, timestamp: Long, outputFilePath: String, metrics: MetricsHelper): Unit = {
+    val conf = new SparkConf()
+      .setAppName("MasterDataProcessorIndexer")
+      .set("spark.redis.host", dataset.datasetConfig.redisDBHost.get)
+      .set("spark.redis.port", String.valueOf(dataset.datasetConfig.redisDBPort.get))
+      .set("spark.redis.db", String.valueOf(dataset.datasetConfig.redisDB.get))
+    val readWriteConf = ReadWriteConfig(scanCount = 1000, maxPipelineSize = 1000)
```

Comment:
Don't hardcode the scan count and other configurations.

Reply:
ok